### PR TITLE
KOGITO-5933: Enable the new Boxed Expression Editor in all Kogito channels

### PR DESCRIPTION
**Pull request by vpellegrino**
Enabling `boxed-expression-component` in all Kogito channels + Distributing `dmn-loader.js` as a single file (w/o chunks) + Reducing the size of distributed files by including only files that are referenced in the HTML header.
vsix_demo

### CODE REVIEW

**Review by yesamer — APPROVED** [review]
@vpellegrino LGTM

**Review by karreiro — COMMENTED** [review]
Thank you, @vpellegrino :)

I've created this PR (https://github.com/vpellegrino/kogito-editors-java/pull/1) to decrease the size of the `.vsix` plugin even more. Please let me know wdyt 🚀 

Thanks again for this PR!

**Comment by vpellegrino** [comment]
> The (build) kogito-tooling-editors-java check has **failed**. Please check [the logs](https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-editors-java.tests.kogito-tooling-editors-java/10/display/redirect).

@karreiro it seems that the removal of unused `appformer-js-monaco assets` is having issues...
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (Remove unused appformer-js-monaco assets) on project kie-wb-common-dmn-webapp-kogito-runtime: An Ant BuildException has occured: Directory does not exist: /home/runner/work/kogito-editors-java/kogito-editors-java/kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/target/kie-wb-common-dmn-webapp-kogito-runtime/org.kie.workbench.common.dmn.showcase.DMNKogitoRuntimeWebapp/appformer-js-monaco
Error:  around Ant part ...<delete>... @ 4:11 in /home/runner/work/kogito-editors-java/kogito-editors-java/kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/target/antrun/build-main.xml
```

**Review by danielzhe — APPROVED** [review]
Good job!

**Comment by danielzhe** [comment]
I was able to build and run locally. 🤔

**Comment by karreiro** [comment]
@vpellegrino I was able to build and run it too (notice that now it's important to build the whole `kogito-editors-java`, I'm not sure if this clue helps tho) :-)

**Comment by vpellegrino** [comment]
> @vpellegrino I was able to build and run it too (notice that now it's important to build the whole `kogito-editors-java`, I'm not sure if this clue helps tho) :-)

Ok, if we are confident that the reported issue is only local to jenkins machine, then I'm ok to merge it

**Review by karreiro — COMMENTED** [review]
@vpellegrino, I've built this PR and tested the runtime web app locally:
Kapture 2021-09-29 at 10 16 13

However, I'm not able to open the DMN editor after I generate the `.vsix` plugin. I'm wondering if we'll need extra changes here https://github.com/kiegroup/kogito-tooling/pull/631. Wdyt?

Thanks again for this PR! 🚀 :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml [hunk 1]
```diff
@@ -791,6 +817,34 @@
...
+          </execution>
+          <execution>
+            <id>Remove unused appformer-js-monaco assets</id>
+            <phase>install</phase>
+            <goals>
+              <goal>run</goal>
+            </goals>
+            <configuration>
+              <tasks>
+                <delete>
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/pom.xml:842]: @vpellegrino I was able to reproduce the build failure locally. It happens if we skip GWT compilation. We may add this change to solve it:

```suggestion
                <delete quiet="true">
```

**Comment by vpellegrino** [comment]
> However, I'm not able to open the DMN editor after I generate the `.vsix` plugin. I'm wondering if we'll need extra changes here [kiegroup/kogito-tooling#631](https://github.com/kiegroup/kogito-tooling/pull/631). Wdyt?
> 
> Thanks again for this PR! rocket :-)

It seems that the issue with generated `.vsix` are related to Monaco removal.
image
I think that the right strategy here, considering the goal of this task, is to revert the commits related to Monaco duplication removal and to go ahead.
FYI, plugin size, on my machine is currently 27MB:
image

Please let me know if that works for you all, so we can proceed with release and unblock https://github.com/kiegroup/kogito-tooling/pull/631

**Review by karreiro — APPROVED** [review]
Thanks for the investigation, @vpellegrino!

**Pull request merged**